src/Servers/IIS/IIS/test/Common.FunctionalTests/OutOfProcess/AspNetCorePortTests.cs

```diff
@@ -84,6 +84,32 @@ public async Task EnvVarInWebConfig_Invalid(TestVariant variant, string port)
             Assert.Equal(HttpStatusCode.BadGateway, response.StatusCode);
         }
 
+        [ConditionalTheory]
+        [MemberData(nameof(TestVariants))]
+        [RequiresNewShim]
+        public async Task ShutdownMultipleTimesWorks(TestVariant variant)
+        {
+            // Must publish to set env vars in web.config
+            var deploymentParameters = Fixture.GetBaseDeploymentParameters(variant);    
+
+            var deploymentResult = await DeployAsync(deploymentParameters);
+
+            var response = await deploymentResult.HttpClient.GetAsync("/Shutdown");
+            response = await deploymentResult.HttpClient.GetAsync("/Shutdown");
+
+            for (var i = 0; i < 5; i++)
+            {
+                // ANCM should eventually recover from being shutdown multiple times.
```

Comment:
Is there a more guaranteed way to do this?

Reply:
I actually think it should be reliable to do one request.